pkg/ingester/ingester.go

```diff
@@ -244,6 +244,15 @@ func (i *Ingester) Shutdown() {
 	i.lifecycler.Shutdown()
 }
 
+// ShutdownHandler triggers the following set of operations in order:
+//     * Change the state of ring to stop accepting writes.
+//     * Flush all the chunks.
+func (i *Ingester) ShutdownHandler(w http.ResponseWriter, r *http.Request) {
+	i.Shutdown()
```

Comment:
So when we call this, we will first look for ingesters to transfer to and then flush the chunks anyways. I don't think we need a `i.Flush()` after. Also please test this in dev once.

Reply:
Ah yes. The flushing was somewhere deep inside the code, found it. Will change and also test it in dev.